src/main/java/org/apache/lucene/analysis/kr/utils/DictionaryUtil.java

```diff
@@ -18,36 +18,127 @@
 	
-	private static Trie<String,WordEntry> dictionary;
+	private static Supplier<Trie<String,WordEntry>> DICTIONARY = Suppliers.memoize(new Supplier<Trie<String, WordEntry>>() {
+        @Override
+        public Trie<String, WordEntry> get() {
+            try {
+                return loadDictionary();
+            } catch (MorphException e) {
+                throw new RuntimeException("Failed to initialize DICTIONARY.", e);
+            }
+        }
+    });
 	
-	private static HashMap josas;
+	private static Supplier<Map<String, String>> JOSAS = Suppliers.memoize(new Supplier<Map<String, String>>() {
+        @Override
+        public Map<String, String> get() {
+            try {
+                return readDict(KoreanEnv.FILE_JOSA);
+            } catch (MorphException e) {
+                throw new RuntimeException("Failed to initialize JOSA dictionary.", e);
+            }
+        }
+    });
 	
-	private static HashMap eomis;
-	
-	private static HashMap prefixs;
-	
-	private static HashMap suffixs;
-	
-	private static HashMap<String,WordEntry> uncompounds;
+	private static Supplier<Map<String, String>> EOMIS = Suppliers.memoize(new Supplier<Map<String, String>>() {
+        @Override
+        public Map<String, String> get() {
+            try {
+                return readDict(KoreanEnv.FILE_EOMI);
+            } catch (MorphException e) {
+                throw new RuntimeException("Failed to initialize EOMI dictionary.", e);
+            }
+        }
+    });
+
+    private static Supplier<Map<String, String>> PREFIXS = Suppliers.memoize(new Supplier<Map<String, String>>() {
+        @Override
+        public Map<String, String> get() {
+            try {
+                return readDict(KoreanEnv.FILE_PREFIX);
+            } catch (MorphException e) {
+                throw new RuntimeException("Failed to initialize PREFIX dictionary.", e);
+            }
+        }
+    });
+
+    private static Supplier<Map<String, String>> SUFFIXS = Suppliers.memoize(new Supplier<Map<String, String>>() {
+        @Override
+        public Map<String, String> get() {
+            try {
+                return readDict(KoreanEnv.FILE_SUFFIX);
+            } catch (MorphException e) {
+                throw new RuntimeException("Failed to initialize SUFFIX dictionary.", e);
+            }
+        }
+    });
+
+	private static Supplier<Map<String, WordEntry>> UNCOMPOUNDS = Suppliers.memoize(new Supplier<Map<String, WordEntry>>() {
+        @Override
+        public Map<String, WordEntry> get() {
+            Map<String, WordEntry> uncompounds = new HashMap<String, WordEntry>();
+            List<String> lines;
+
+            try {
+                lines = FileUtil.readLines(KoreanEnv.getInstance().getValue(KoreanEnv.FILE_UNCOMPOUNDS),"UTF-8");
+            } catch (Exception e) {
+                throw new RuntimeException("Failed to initialize UNCOMPOUNDS dictionary.", e);
+            }
+
+            for(String compound: lines) {
```

Comment:
I just moved that block, so I tried to change as little as possible. If I start changing, I will probably change every line in that code. There is StringBuffer usage everywhere!!!